[KEYCLOAK-16098] (Part of KEYCLOAK-15692) Documentation for the upgrade to Wildfly 21

iankko opened the pull request:
[KEYCLOAK-16098] Documentation for the upgrade to Wildfly 21
    
    Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

This is now ready for review. Related PRs:

- [Code change](https://github.com/keycloak/keycloak/pull/7463)
- [Website change](https://github.com/keycloak/keycloak-web/pull/146)

Note: This change intentionally wasn't rebased to current / latest `master`, so it can be verified / tested till the fix for KEYCLOAK-16122 issue is found (will rebase it once its fixed)

@abstractj @andymunro @hmlnarik @mhajas @martin-kanis @mposolda @pedroigor @pskopek @slaskawi @stianst @vmuzikar @vramik PTAL once got a chance

Thanks!
Jan

Commit:
[KEYCLOAK-16098] Documentation for the upgrade to Wildfly 21

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

andymunro commented on upgrading/topics/keycloak/changes.adoc:
```diff
@@ -2,6 +2,18 @@
+  The dependencies were updated to the versions used by Wildfly 21 server. For example, Infinispan is now 11.0.4.Final.
+
+Configuration changes::
+  There are few configuration changes in the `standalone(-ha).xml` and `domain.xml` files. You should follow the <<_install_new_version>>
```
Generally avoid "there are" which makes no sense grammatically, because "there is not the subject of the sentence."
You can usually rephrase using the verb "exists"
So please make it "A few configuration changes exist in..."

iankko replied:
Thanks for the suggestion, fixed in latest version.

andymunro commented on upgrading/topics/keycloak/changes.adoc:
```diff
@@ -257,7 +269,7 @@ we cannot assume which one will be used to fulfill the auth request so the reque
 
 The {project_name} server was upgraded to use Wildfly 15 as the underlying container. This does not directly involve any
-specific {project_name} server functionality, but there are few changes related to the migration, which worth mentioning.
+specific {project_name} server functionality, but there are few changes related to the migration, which are worth mentioning.
```
change this to "but a few configuration changes related to..."

iankko replied:
Corrected per the suggestion. Thanks, Andy

andymunro reviewed (changes requested):
A few changes

iankko commented:
> A few changes

@andymunro Thanks for the review, Andy. Corrected the PR per your suggestions (for the `[tT]here is` / `[tT]there are` instances while was at this corrected all of them, where I had the feeling they are prone to the very same issue. Please take a 2nd look & let me know if they are OK)

Also, added one sentence to the WF 21 upgrade yet (that the `object-memory` element was replaced with the `heap-memory` one), I forgot to add before.

PTAL & let me know, if any further changes are needed yet

Thank you
Jan
